app/screens/settings/display_theme/display_theme.tsx

```diff
@@ -26,48 +27,54 @@ type DisplayThemeProps = {
 const DisplayTheme = ({allowedThemeKeys, componentId, currentTeamId, currentUserId}: DisplayThemeProps) => {
     const serverUrl = useServerUrl();
     const theme = useTheme();
-    const initialTheme = useRef(theme);
-    const [newTheme, setNewTheme] = useState<string | undefined>(undefined);
+    const [currentTheme, setCurrentTheme] = useState(theme);
+    const [customTheme, setCustomTheme] = useState(theme.type?.toLowerCase() === 'custom' ? theme : undefined);
 
     const close = () => popTopScreen(componentId);
 
-    const setThemePreference = useCallback(() => {
-        const allowedTheme = allowedThemeKeys.find((tk) => tk === newTheme);
-        const themeJson = Preferences.THEMES[allowedTheme as ThemeKey] || initialTheme.current;
+    const setThemePreference = useCallback(async (theTheme: string) => {
+        const allowedTheme = allowedThemeKeys.find((tk) => tk === theTheme);
+        const themeJson = Preferences.THEMES[allowedTheme as ThemeKey] || customTheme;
 
         const pref: PreferenceType = {
             category: Preferences.CATEGORIES.THEME,
             name: currentTeamId,
             user_id: currentUserId,
             value: JSON.stringify(themeJson),
         };
-        savePreference(serverUrl, [pref]);
-    }, [allowedThemeKeys, currentTeamId, currentUserId, serverUrl, newTheme]);
+        await savePreference(serverUrl, [pref]);
+
+        setCurrentTheme(themeJson);
+
+    }, [allowedThemeKeys, currentTeamId, currentUserId, serverUrl, customTheme]);
+
+    const handleSelectTheme = usePreventDoubleTap(useCallback((themeSelected: string) => {
+        setThemePreference(themeSelected);
+    }, [setThemePreference]));
```

Comment:
This is the actual fix, adding usePreventDoubleTap(), which disallow the user from tapping the themes rapidly